packages/react-native/Libraries/Text/TextInput/RCTBaseTextInputView.m

```diff
@@ -27,6 +31,31 @@ @implementation RCTBaseTextInputView {
   BOOL _didMoveToWindow;
 }
 
+- (void)reactUpdateResponderOffsetForScrollView:(RCTScrollView *)scrollView
+{
+  if (![self isDescendantOfView:scrollView]) {
+    // View is outside scroll view
+    return;
+  }
+
+  UITextRange *selectedTextRange = self.backedTextInputView.selectedTextRange;
+  UITextSelectionRect *selection = [self.backedTextInputView selectionRectsForRange:selectedTextRange].firstObject;
+  if (selection == nil) {
+    // No active selection or caret - fallback to entire input frame
+    scrollView.firstResponderFocus = [self convertRect:self.bounds toView:nil];
+  } else {
+    // Focus on text selection frame
+    BOOL isMultiline = [self.backedTextInputView isKindOfClass:[UITextView class]];
+    CGRect focusRect = CGRectMake(
+      selection.rect.origin.x,
+      selection.rect.origin.y,
+      selection.rect.size.width,
+      selection.rect.size.height + (isMultiline ? 0 : kSingleLineKeyboardBottomOffset)
+    );
```

Comment:
```suggestion
    CGRect focusRect = selection.rect;
    focusReact.size.height += (isMultiline ? 0 : kSingleLineKeyboardBottomOffset)
```

Reply:
Thanks for the suggestion! I've just incorporated that and reworked that section more broadly to reduce duplication and try and make it all a bit more obvious what going on.